### ISSUE TITLE
Coupling with Energy Model (Task 1)

### ISSUE THREAD

**Issue by colinsheppard** [LBNL-UCB-STI/beam#940]
This is the first task in an epic of tasks around enabling more detailed energy modeling with BEAM by coupling to a separate model.

First step is to create a new output file, we'll control when this writes using:

beam.outputs.writeLinkTraversalsInterval 

With a default value of 0 (so no output is ever written by default).

Then, similar to e.g. "beam.outputs.writeEventsInterval" we use the interval to decide which iteration we should write these outputs.

Then create a CSV output file during the appropriate iteration where we write the following data for each link traversal (i.e. for each vehicle that moves through each link):

linkId,linkType,averageSpeed,freeFlowSpeed,linkEnterTime,vehicleId,vehicleType,turnAtLinkEnd,numberOfStops

Further clarification on the less obvious of these:

linkType -- this needs to be functional class of the link (highway, arterial, local, etc.) The data from this should come from the physsim network (see attribute "type"). 

averageSpeed - in m/s the actual speed to traverse the link as provided in BeamLeg.travelPath.linkTravelTime

freeFlowSpeed, can come also from physsim network

turnAtLinkEnd - one of "L", "R", "S", or "NA" for left, right, straight, or not applicable. Difference between S and NA is we use S to denote that we have chosen to go straight through an intersection and NA if there was no intersection.

numberOfStops - for now, just put 0 if turnAtLinkEnd is NA and 1 otherwise. In another task we will enhance this to add realism.

**Comment by bhavyalatha26** [comment]
@colinsheppard I am stuck at two points : 
- How  to fetch a beamLeg/trip reference in beam sim class 
- On factor should we determine , whether "turnAtLinkEnd" should be L / R / S / NA

**Comment by bhavyalatha26** [comment]
@colinsheppard , the link attributes "type" and "origid" are now deprecated. So we may not be able to get linkType using the attribute

**Comment by bhavyalatha26** [comment]
linkType to be replace with **linkCapacity**

**Comment by bhavyalatha26** [comment]
Test spec complete . Waiting on some issue reported by @rajnikantsh on sf-light scenario running for longer time
